ibridges/search.py

```diff
@@ -0,0 +1,54 @@
+""" Data query
+"""
+from typing import Optional
+
+from ibridges.irodsconnector import keywords as kw
+from ibridges import Session
+
+
+def search(session: Session, path: Optional[str] = None, checksum: Optional[str] = None,
+               key_vals: Optional[dict] = None) -> list[tuple[str, str, str]]:
+    """Retrieves all collections and data objects (the absolute collection path,
+    data object or collection name) to the given user-defined and system metadata.
+    By Default all accessible collections and data objects will be returned.
+    Wildcard: %
+
+    Parameters
+    ----------
+    path: str
+        (Partial) path
+    checksum: str
+        (Partial) checksum
+    key_vals : dict
+        Attribute name mapping to values.
+
+    Returns
+    -------
+    list: [(Collection name, Object name, checksum)]
+
+    """
+    if path is None and checksum is None and key_vals is None:
+        raise ValueError("QUERY DATA: No query criteria set.")
```

Comment:
I would suggest a bit more feedback to the user here. Perhaps something like:

Error while searching in the metadata: No query criteria set. Please supply either a path, checksum or key_vals.

Now that I think about it, maybe we can also just set path to session.home? Then you just get all objects in the current working directory.

Reply:
I also thought to give that as a default. However, when a home collection gets very large, this search can take some time. That's why I decided to throw an error.
@maartenschermer Do you have a strong opinion about this?